store/src/main/java/org/apache/rocketmq/store/ConsumeQueue.java

```diff
@@ -151,7 +151,7 @@ public void recover() {
         }
     }
 
-    public long getOffsetInQueueByTime(final long timestamp) {
+    public long getOffsetInQueueByTime(final long timestamp, boolean isGetTimeLast) {
         MappedFile mappedFile = this.mappedFileQueue.getMappedFileByTime(timestamp);
```

Comment:
It is not reliable to get mapped file by modification time, as if the dispatch service fall behind too much, it may get the wrong mapped file.
How about checking each file's first message and last message it pointed to?
This may cause performance loss, but it is acceptable, for such api is used only for some specific diagnoses. It is rarely used, but when using it, it'd better to return the exactly correct result.

Reply:
@lindzh please consider such problem

Reply:
Thanks for you advice,I will check it later.If exist i will change it in the next commit.